src/strncasecmp.rs

```diff
@@ -0,0 +1,53 @@
+//! Rust implementation of C library function `strncasecmp`
+//!
+//! Licensed under the Blue Oak Model Licence 1.0.0
+
+use crate::{CChar, CInt};
+
+/// Rust implementation of C library function `strncasecmp`. Passing NULL
+/// (core::ptr::null()) gives undefined behaviour.
+#[cfg_attr(feature = "strncasecmp", no_mangle)]
+pub unsafe extern "C" fn strncasecmp(s1: *const CChar, s2: *const CChar, n: usize) -> CInt {
+	for i in 0..n as isize {
```

Comment:
```suggestion
	for i in 0..n {
		let s1_i = s1.add(i);
		let s2_i = s2.add(i);
```

Doesn't make much difference in practice, but we can avoid the cast I think.

Reply:
sure, it was mostly copy-paste from `strncmp`